File: java-source/src/main/java/net/corda/training/MessageContract.java
package net.corda.training;

import net.corda.core.contracts.*;
import net.corda.core.transactions.LedgerTransaction;
import java.util.Collections;
import static net.corda.core.contracts.ContractsDSL.requireSingleCommand;
import static net.corda.core.contracts.ContractsDSL.requireThat;

/* Contract */
public class MessageContract implements Contract {

    static final String ID = "net.corda.training.MessageContract";
    static class SendMessage extends TypeOnlyCommandData{} // TODO: Is this ok?

    @Override
    public void verify(LedgerTransaction tx) {
        final CommandWithParties<SendMessage> command = requireSingleCommand(tx.getCommands(), SendMessage.class);
        requireThat( require -> {
            require.using("There should one output state.", tx.getOutputStates().size() == 1);
            final MessageState outputState = tx.outputsOfType(MessageState.class).get(0);
            require.using("The party sending the message must sign the SendMessage transaction.", (command.getSigners().equals(Collections.singletonList(outputState.origin.getOwningKey()))));
            // require.using("There should be no input state.", tx.getInputStates().isEmpty());
            return null;
        });
    }
}

File: kotlin-source/src/main/kotlin/net/corda/hello/MessageContract.kt
package net.corda.hello

import net.corda.core.contracts.Contract
import net.corda.core.contracts.TypeOnlyCommandData
import net.corda.core.contracts.requireSingleCommand
import net.corda.core.contracts.requireThat
import net.corda.core.transactions.LedgerTransaction

/* Contract */
/* TODO(#2): Add a constraint that ensures that our transaction has no input state! */
class MessageContract: Contract {

    companion object { const val ID = "net.corda.hello.MessageContract" }
    class SendMessage : TypeOnlyCommandData()

    override fun verify(tx: LedgerTransaction) = requireThat {
        val command = tx.commands.requireSingleCommand<SendMessage>()
        "There should be one output state." using (tx.outputStates.size == 1)
        val outputState = tx.outputStates.single() as MessageState
        "The party sending the message must sign the SendMessage transaction." using (command.signers == listOf(outputState.origin.owningKey))

        // Add this
        // "There should be no input state." using tx.inputStates.isEmpty()
    }
}